pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test: TestLogic_synthetic_privileges failed

pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test.TestLogic_synthetic_privileges [failed](https://mesolite.cluster.engflow.com/invocations/default/56985aab-c148-4a56-932f-0094fed89725?testReportRun=1&testReportShard=35&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9mYWtlZGlzdC12ZWMtb2ZmOmZha2VkaXN0LXZlYy1vZmZfdGVzdA==) on master @ [d9dbc2ef1e9db9491e7af7530d148df85b4d6214](https://github.com/cockroachdb/cockroach/commits/d9dbc2ef1e9db9491e7af7530d148df85b4d6214):


```
=== RUN   TestLogic_synthetic_privileges
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLogic_synthetic_privileges100972241
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[22:00:46] setting distsql_workmem='12277B';
[22:00:48] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 64 statements
    logic.go:4308: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_synthetic_privileges100972241
--- FAIL: TestLogic_synthetic_privileges (3.62s)
=== RUN   TestLogic_synthetic_privileges/all_does_not_include_nosqllogin
    logic.go:2205: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:394: 
        expected success, but found
        read tcp 127.0.0.1:60610->127.0.0.1:39921: read: connection reset by peer
[22:00:48] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges with config fakedist-vec-off: 81 tests, 1 failures
[22:00:49] --- total progress: 81 statements
--- total: 81 tests, 1 failures
    --- FAIL: TestLogic_synthetic_privileges/all_does_not_include_nosqllogin (0.06s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=35</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_synthetic_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-38902

pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test.TestLogic_synthetic_privileges [failed](https://tanzanite.cluster.engflow.com/invocations/default/22d7d37f-9dfb-4e55-8b11-dffe4e6c749b?testReportRun=2&testReportShard=35&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9mYWtlZGlzdC12ZWMtb2ZmOmZha2VkaXN0LXZlYy1vZmZfdGVzdA==) on master @ [d146ecff6f687e438706cf63591cafca60cc116d](https://github.com/cockroachdb/cockroach/commits/d146ecff6f687e438706cf63591cafca60cc116d):


```
=== RUN   TestLogic_synthetic_privileges
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLogic_synthetic_privileges2319298584
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[06:45:38] --- progress: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 65 statements
    logic.go:4308: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_synthetic_privileges2319298584
--- FAIL: TestLogic_synthetic_privileges (3.52s)
=== RUN   TestLogic_synthetic_privileges/all_does_not_include_nosqllogin
    logic.go:2205: 
         
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:394: 
        expected success, but found
        read tcp 127.0.0.1:51856->127.0.0.1:46169: read: connection reset by peer
[06:45:38] --- done: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges with config fakedist-vec-off: 81 tests, 1 failures
[06:45:39] --- total progress: 81 statements
--- total: 81 tests, 1 failures
    --- FAIL: TestLogic_synthetic_privileges/all_does_not_include_nosqllogin (0.06s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=35</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_synthetic_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/synthetic_privileges
+++ b/pkg/sql/logictest/testdata/logic_test/synthetic_privileges
@@ -388,7 +388,8 @@ GRANT SYSTEM ALL TO testuser
 statement ok
 CANCEL SESSION (SELECT session_id FROM [SHOW SESSIONS] WHERE user_name = 'testuser')
 
-user testuser
+# Force a new session to be created, causing a connection attempt.
+user testuser nodeIdx=0 newsession
 
 statement ok
 SELECT 1

